From: Iain Lane <laney@debian.org>
Date: Fri, 6 Nov 2020 13:33:13 +0000
Subject: Allow background-color and color on VteTerminal widgets to be
 customised via the theme

Yaru and Ambiance/Radiance need this. Move the CSS rules to the fallback
style class so that they can be overridden."

Forwarded: not-needed
---
 src/vtegtk.cc | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/vtegtk.cc b/src/vtegtk.cc
index f7a65f7..fb3d5bc 100644
--- a/src/vtegtk.cc
+++ b/src/vtegtk.cc
@@ -2024,14 +2024,16 @@ vte_terminal_class_init(VteTerminalClass *klass)
         klass->priv->fallback_style_provider = GTK_STYLE_PROVIDER (gtk_css_provider_new ());
         /* Some themes don't define text_view_bg */
         gtk_css_provider_load_from_data (GTK_CSS_PROVIDER (klass->priv->fallback_style_provider),
-                                         "@define-color text_view_bg @theme_base_color;",
+                                         "@define-color text_view_bg @theme_base_color;\n"
+                                         "VteTerminal, " VTE_TERMINAL_CSS_NAME " {\n"
+                                         "background-color: @text_view_bg;\n"
+                                         "color: @theme_text_color;\n"
+                                         "}\n",
                                          -1, NULL);
         klass->priv->style_provider = GTK_STYLE_PROVIDER (gtk_css_provider_new ());
         gtk_css_provider_load_from_data (GTK_CSS_PROVIDER (klass->priv->style_provider),
                                          "VteTerminal, " VTE_TERMINAL_CSS_NAME " {\n"
                                          "padding: 1px 1px 1px 1px;\n"
-                                         "background-color: @text_view_bg;\n"
-                                         "color: @theme_text_color;\n"
                                          "}\n",
                                          -1, NULL);
 
